agent/eni/watcher_linux.go

```diff
@@ -182,11 +169,10 @@ func (udevWatcher *UdevWatcher) eventHandler(ctx context.Context) {
 					log.Warnf("Udev watcher event-handler: error obtaining MACAddress for interface %s", netInterface)
 					continue
 				}
-				udevWatcher.state.AddDeviceWithMACAddress(netInterface, macAddress)
+				udevWatcher.state.HandleENIEvent(macAddress)
 			case udevRemoveEvent:
 				netInterface := event.Env[udevInterface]
 				log.Debugf("Udev watcher event-handler: remove interface: %s", netInterface)
-				udevWatcher.state.RemoveDevice(netInterface)
```

Comment:
Then, we should just get rid of `case udevRemoveEvent` and handle only `udevAddEvent`, in which case this doesn't event need to be a `switch-case` block